sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemPropertiesTable.java

```diff
@@ -0,0 +1,174 @@
+    druidServers.forEachRemaining(discoveryDruidNode -> {
+      final DruidNode druidNode = discoveryDruidNode.getDruidNode();
+      final Map<String, String> propertiesMap = getProperties(druidNode);
+      if (serverToPropertiesMap.containsKey(druidNode.getHostAndPortToUse())) {
+        Pair<StringBuilder, Stream<Object[]>> pair = serverToPropertiesMap.get(druidNode.getHostAndPortToUse());
+        pair.lhs.append(StringUtils.format(",%s", discoveryDruidNode.getNodeRole().getJsonName()));
+      } else {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(StringUtils.format("%s", discoveryDruidNode.getNodeRole().getJsonName()));
+        serverToPropertiesMap.put(druidNode.getHostAndPortToUse(), Pair.of(builder, propertiesMap.entrySet().stream()
+                                                                                                              .map(entry -> new Object[]{
+                                                                                                                    druidNode.getServiceName(),
+                                                                                                                    druidNode.getHostAndPortToUse(),
+                                                                                                                    entry.getKey(),
+                                                                                                                    entry.getValue()
+                                                                                                              })
+            )
+        );
+      }
+    });
+    return Linq4j.asEnumerable(serverToPropertiesMap.values().stream().flatMap(pair -> pair.rhs.map(entry -> new Object[]{entry[0], entry[1], pair.lhs.toString(), entry[2], entry[3]})).collect(Collectors.toList()));
+  }
+
+  private Map<String, String> getProperties(DruidNode druidNode)
+  {
+    final String url = druidNode.getUriToUse().resolve("/status/properties").toString();
+    try {
+      final Request request = new Request(HttpMethod.GET, new URL(url));
+      final StringFullResponseHolder response;
+      response = httpClient
+          .go(request, new StringFullResponseHandler(StandardCharsets.UTF_8))
+          .get();
+
+      if (response.getStatus().getCode() != HttpServletResponse.SC_OK) {
+        throw new RE(
+            "Failed to get properties from node at [%s]. Error code [%d], description [%s].",
+            url,
+            response.getStatus().getCode(),
+            response.getStatus().getReasonPhrase()
+        );
+      }
+      return jsonMapper.readValue(
+          response.getContent(),
+          new TypeReference<Map<String, String>>()
+          {
+          }
+      );
+    }
+    catch (IOException | InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+    catch (ExecutionException e) {
```

Comment:
Let's club these to just catch `Exception` and throw a `DruidException` since this will be sent back to the user.